### PR TITLE
Allow tags to be set on a node during bootstrap

### PR DESCRIPTION
**Pull request by swalberg**
I can set attributes, environment, and run_list during a `knife bootstrap`, but why not tags?

Some of our recipes are driven off of tags, such as for sudo access or
membership in a load balancing pool.

If this patch is accepted then I will add the corresponding feature to
[knife-vsphere](https://github.com/ezrapagel/knife-vsphere).

### CODE REVIEW

File: spec/unit/knife/core/bootstrap_context_spec.rb [hunk 1]
```diff
@@ -97,10 +97,10 @@
     end
   end
 
-  describe "when JSON attributes are given" do
-    let(:config) { {:first_boot_attributes => {:baz => :quux}} }
+  describe "when tags are not given" do
```
Comment (coderanger) [inline spec/unit/knife/core/bootstrap_context_spec.rb]: This should be a new test, not a replacement (unless there is a pressing reason to not test the old behavior).

**Comment by coderanger** [comment]
:-1: overall, tags should probably be deprecated, no reason to add new footguns :-(

File: spec/unit/knife/core/bootstrap_context_spec.rb (hunk 1 above)
Comment (swalberg) [inline spec/unit/knife/core/bootstrap_context_spec.rb]: That's a mistake on my part, my intention was to add a new test.

**Comment by swalberg** [comment]
I can understand the logic behind disliking anything to do with an array. But tags are a current feature and a helpful one at that. In a future world, should tags be dropped or just fixed?

**Comment by thommay** [comment]
@swalberg I'm pretty sure we're not intending to deprecate tags; as you say they're a useful and an (admittedly not widely) used feature.

**Comment by ranjib** [comment]
:+1:   
Adding to what @thommay said, tags are useful and I love them. 
But I think we should seriously revisit the bootstrapping mechanism, instead of changing the CLI interface and the erb rendering logic, may be consume a yaml or othet config with all the tags, hooks etc. But this looks fine to me, in context current bootstrap mechanism.

**Comment by thommay** [comment]
I'm :+1: on this, but thoroughly agree with @ranjib wrt overhauling bootstrapping.

**Comment by stevendanna** [comment]
:+1:

**Comment by smurawski** [comment]
:+1:

**Comment by lamont-granquist** [comment]
yeah, so I guess :+1: we can already set these with normal attrs, but clearly that isn't obvious to everyone...

**Comment by thommay** [comment]
@chef/client-maintainers can we get some more reviews of this please? We have an outstanding -1 here, so more opinions welcome.

**Comment by mwrock** [comment]
:+1:

**Comment by danielsdeleo** [comment]
:+1: 

@coderanger if you want to get rid of tags, make an RFC? Otherwise it seems to me we should support using them rather than only having them be usable in some places but not others. Even if we did get rid of them (hypothetically), we couldn't do that until 13.0 and it wouldn't be bad to let people use them (with an appropriate warning) until then.

**Comment by btm** [comment]
:+1: agree with @danielsdeleo, this isn't the correct forum to discuss deprecating tags.

**Pull request merged**